### PR TITLE
add abstract setup/teardown executor hooks

### PR DESCRIPTION
**Pull request by venkatBala**
- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

### CODE REVIEW

**Review by cjao — DISMISSED** [review]

File: covalent/executor/base.py [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import subprocess
 import tempfile
 from abc import ABC, abstractmethod
+from ast import Call
```
Comment (cjao) [inline covalent/executor/base.py:31]: What is this used for?

**Review by cjao — CHANGES_REQUESTED** [review]

File: covalent/executor/base.py (hunk 1 above)
Comment (venkatBala) [reply covalent/executor/base.py:31]: ah my bad... bad typing import

**Review by cjao — APPROVED** [review]

File: covalent/executor/base.py [hunk 2]
```diff
@@ -493,6 +510,14 @@ async def execute(
 
         return (result, stdout.getvalue(), stderr.getvalue())
 
+    async def setup(self):
```
Comment (cjao) [inline covalent/executor/base.py:513]: Are you planning to integrate `setup` and `teardown` with `execute()`?

**Review by cjao — CHANGES_REQUESTED** [review]

**Comment by wjcunningham7** [comment]
This pull request is stale because it has been in review for at least three days. It will be closed in two more days.

File: covalent/executor/base.py (hunk 2 above)
Comment (wjcunningham7) [reply covalent/executor/base.py:513]: the dispatcher should call setup and teardown before/after execute, if they exist

**Review by cjao — APPROVED** [review]

**Pull request merged**